### PR TITLE
Reset passwd validation

### PR DESCRIPTION
**Pull request by msaianand**
https://trello.com/c/uOetaujz/56-front-end-validation-for-reset-password

### CODE REVIEW

**Comment by msaianand** [comment]
[Front end validation for reset password](https://trello.com/c/uOetaujz/56-front-end-validation-for-reset-password)

File: pom.xml [hunk 1]
```diff
@@ -173,6 +173,8 @@
             <groupId>org.onedatashare.module</groupId>
             <artifactId>globus-api</artifactId>
             <version>1.1-SNAPSHOT</version>
+            <scope>system</scope>
+            <systemPath>${project.basedir}/lib/globus-api-1.1-SNAPSHOT.jar</systemPath>
```
Comment (aashish-jain) [inline pom.xml:177]: Please remove these lines from pom.xml

File: src/main/java/org/onedatashare/server/service/UserService.java [hunk 2]
```diff
@@ -154,17 +155,28 @@ public GlobusClient getGlobusClientFromUser(User user){
 
   public Mono<String> resetPasswordWithOld(String cookie, String oldpassword, String newpassword, String passwordConfirm){
     return getLoggedInUser(cookie).flatMap(user-> {
+      //OldPwdMatchingException oldPwd = new OldPwdMatchingException()
+      //System.out.println("oldpassword "+oldpassword);
```
Comment (aashish-jain) [inline src/main/java/org/onedatashare/server/service/UserService.java]: Remove theses comments

File: src/main/java/org/onedatashare/server/service/UserService.java [hunk 3]
```diff
@@ -154,17 +155,28 @@ public GlobusClient getGlobusClientFromUser(User user){
...
+      //OldPwdMatchingException oldPwd = new OldPwdMatchingException()
+      //System.out.println("oldpassword "+oldpassword);
       if(!newpassword.equals(passwordConfirm)){
         ODSLoggerService.logError("Passwords don't match.");
-        return Mono.error(new Exception("Passwords don't match."));
+        throw new  OldPwdMatchingException("Passwords don't match.");
       }else if(!user.checkPassword(oldpassword)){
         ODSLoggerService.logError("Old Password is incorrect.");
-        return Mono.error(new Exception("Old Password is incorrect."));
+       // return Mono.error(new Exception("Old Password is incorrect."));
```
Comment (aashish-jain) [inline src/main/java/org/onedatashare/server/service/UserService.java]: Remove this comment

File: src/main/java/org/onedatashare/server/service/UserService.java [hunk 4]
```diff
@@ -154,17 +155,28 @@ public GlobusClient getGlobusClientFromUser(User user){
...
         userRepository.save(user).subscribe();
         ODSLoggerService.logInfo("Password reset for user " + user.getEmail() + " successful.");
         return Mono.just(user.getHash());
+        }
+        catch (RuntimeException e)
+        {
+          throw  new OldPwdMatchingException(e.getMessage());
+        }
+
+
```
Comment (aashish-jain) [inline src/main/java/org/onedatashare/server/service/UserService.java]: The indentation doesn't look proper. Please fix it.

**Comment by aashish-jain** [comment]
Changes look good. Please address the mentioned issues.

File: src/main/java/org/onedatashare/server/service/UserService.java (hunk 2 above)
Comment (msaianand) [reply src/main/java/org/onedatashare/server/service/UserService.java]: removed comments

File: src/main/java/org/onedatashare/server/service/UserService.java (hunk 3 above)
Comment (msaianand) [reply src/main/java/org/onedatashare/server/service/UserService.java]: removed comments

File: src/main/java/org/onedatashare/server/service/UserService.java (hunk 4 above)
Comment (msaianand) [reply src/main/java/org/onedatashare/server/service/UserService.java]: fixed indentation

**Comment by LinusCastelino** [comment]
Merge conflicts exist and there are some minor concerns. Will hold merging until next meeting.

**Comment by LinusCastelino** [comment]
Merge conflicts fixed and added additional UI fixes.

Please review and approve @yinyifu @aashish-jain @KiranPrabhakar135 @dnidhi2710

**Review by aashish-jain — APPROVED** [review]
Looks good

**Pull request merged**